src/Tinfoil/AEAD/AESGCM/Data.hs

```diff
@@ -34,7 +36,7 @@ newtype RandomField =
 -- startup.
 newtype InvocationCount =
   InvocationCount {
-    unInvocationCount :: Integer
+    unInvocationCount :: Word64
```

Comment:
Yep that was the other option I considered, used a `Word64` at @erikd-ambiata's suggestion to avoid the `fromIntegral`. I do agree that a `Word32` would be clearer though, might go with that.